### DISCUSSION TITLE
Feature req: Set `session.user.id` automatically when using a database adapter

### DISCUSSION THREAD

**Discussion by jasongitmail** [nextauthjs/next-auth#8911] [Ideas]
### Goals

Automatically make `session.user.id` available on the session when a database adapter exists in the config, irrespective of whether using a `database` or `jwt` session strategy. It should contain the `id` from the `user` database table.

### Background

Currently you have to add this manually, like this for a database session:

```ts
async session({ session, user }) {
  if (session?.user) {
    // @ts-expect-error Property id does not exist
    session.user.id = user.id;
  }

  return session;
},
```

but then this also breaks if changing to a session.strategy of `jwt`.

### Proposal

This is so common, it should be default to create a better developer experience, for both sessions stored in the database or jwt, so long as a database adapter exists.

**Comment by FunMiles** [comment]
I agree. I've written  a variant, as I have been switching between database and jwt strategies:
```js
session({ session, token, user }) {
   if (session?.user) {
     // @ts-expect-error Property id does not exist
     session.user.id = user?.id || token?.sub;
   }
   return session
}
```

**Reply by 0xMihir** [reply]
Also, the session user id outputs an integer, not a string.
